WordPress/gutenberg
# Avatar: Fix application of borders

Fixes: https://github.com/WordPress/gutenberg/issues/47408

## What?

Ensures border classes and styles are correctly applied to the avatar block's markup.

_Note: The original issue notes this as an issue for classic themes however it should be a problem for block themes as well._ 

## Why?

The avatar block adopts border supports and skips the serialization of the support classes and styles so they can be applied to an inner image element. Given the block skips serialization it must handle the application of all border configurations and it is currently missing the handling of split or "individual" borders.

## How?
- Replaces the ad-hoc class and style generation with an approach leveraging the style engine
- Creates a helper function for the Avatar block to generate the classes and styles for border support, then leverages that.
- Adds a `box-sizing: border-box` reset to the block's `img` element so borders don't break out of the avatar's sizing.

## Testing Instructions

1. Edit a post, adding an avatar block
2. Selec the block and configure a flat border (i.e. uniform width, color, and style) for all sides
3. Save the post and confirm the correct styles on the frontend
4. Back in the editor, tweak the border styles so they are different per side.
5. Save and verify the individual split borders are reflected correctly on the frontend.
6. Switch back to the editor and toggle on the block's "Link to user profile" option
7. Save and confirm borders are applied correctly still. Repeat for the flat borders.

## Screenshots or screencast 

| Classic Theme (TwentyTwenty) | Block Theme (TwentyTwentyThree) |
|---|---|
| <video src="https://user-images.githubusercontent.com/60436221/215984731-2043d13f-e584-4494-bc23-d3db40dfbd1f.mp4" /> | <video src="https://user-images.githubusercontent.com/60436221/215986192-76ef7e6c-8895-47a3-8e08-4e2df4120d59.mp4" /> |
